build_system/seqan3-config.cmake

```diff
@@ -615,13 +607,20 @@ if (NOT ${FIND_NAME}_FIND_QUIETLY)
 endif ()
 
 separate_arguments (SEQAN3_CXX_FLAGS_LIST UNIX_COMMAND "${SEQAN3_CXX_FLAGS}")
-string(REGEX REPLACE ";( )*[-/]D" ";" SEQAN3_DEFINITIONS_LIST ";${SEQAN3_DEFINITIONS};")
 
-add_library (seqan3::seqan3 INTERFACE IMPORTED)
-set_property (TARGET seqan3::seqan3 APPEND PROPERTY INTERFACE_COMPILE_DEFINITIONS ${SEQAN3_DEFINITIONS_LIST})
-set_property (TARGET seqan3::seqan3 APPEND PROPERTY INTERFACE_COMPILE_OPTIONS ${SEQAN3_CXX_FLAGS_LIST})
-set_property (TARGET seqan3::seqan3 APPEND PROPERTY INTERFACE_LINK_LIBRARIES "${SEQAN3_LIBRARIES}")
-set_property (TARGET seqan3::seqan3 APPEND PROPERTY INTERFACE_INCLUDE_DIRECTORIES "${SEQAN3_INCLUDE_DIRS}")
+add_library (seqan3_seqan3 INTERFACE)
+target_compile_definitions (seqan3_seqan3 INTERFACE ${SEQAN3_DEFINITIONS})
+target_compile_options (seqan3_seqan3 INTERFACE ${SEQAN3_CXX_FLAGS_LIST})
+target_link_libraries (seqan3_seqan3 INTERFACE "${SEQAN3_LIBRARIES}")
+# include seqan3/include/ as -I, because seqan3 should never produce warnings.
+target_include_directories (seqan3_seqan3 INTERFACE "${SEQAN3_BASEDIR}")
+# include everything except seqan3/include/ as -isystem, i.e.
+# a system header which suppresses warnings of external libraries.
+target_include_directories (seqan3_seqan3 SYSTEM INTERFACE "${SEQAN3_INCLUDE_DIRS}")
+add_library (seqan3::seqan3 ALIAS seqan3_seqan3)
+
+# propagate SEQAN3_BASEDIR into SEQAN3_INCLUDE_DIRS
+set (SEQAN3_INCLUDE_DIRS ${SEQAN3_BASEDIR} ${SEQAN3_INCLUDE_DIRS})
```

Comment:
I didn't rename, because I would have broken this. If you don't have any problem with it, I will change it.